modules/tiled-gallery/tiled-gallery.php

```diff
@@ -103,7 +104,17 @@ public function gallery_shortcode( $val, $atts ) {
 
 		$this->set_atts( $atts );
 
-		$attachments = $this->get_attachments();
+		/**
+	         * Filters array of attachments for Tiled Gallery template.
+	         *
+	         * @module tiled-gallery
+	         *
+	         * @since 3.9.0
+	         *
+	         * @param array Array of attachments.
+	         * @param array Array of gallery shortcode attributes.
+	         */
+		$attachments = apply_filters( 'jetpack_gallery_attachments', array(), $this->atts );
```

Comment:
You are right. I was a bit in hurry... I also need to reconsider the whole approach, since I am thinking about adding similar filters to WP core. I need a day or two. Thanks!